### ISSUE TITLE
Type of dataclass Field.type incorrectly reported as Any | str

### ISSUE THREAD

**Issue by adampauls** [microsoft/pyright#8429]
**Describe the bug**
Field.type on in `dataclasses` was recently [changed](https://github.com/python/typeshed/pull/12082) to have type `Type[_T] | str | Any`. However, pyright seems to think the type is `Any | str`. 

**Code or Screenshots**
```python
from dataclasses import dataclass, fields

@dataclass
class Foo:
    x: int

for field in fields(Foo):
    # reveal_type(field.type)  # Type of "field.type" is "Any | str"
    print(type(field.type)) # prints "<class 'type'>"


**VS Code extension or command-line**
Command line 1.1.371.

**Comment by erictraut** [comment]
Pyright is working as intended here, so I don't consider this a bug.

The `fields` function is defined as follows in typeshed:

```python
def fields(class_or_instance: DataclassInstance | type[DataclassInstance]) -> tuple[Field[Any], ...]: ...
```

Not surprisingly, mypy's behavior is the same here.

**Comment by adampauls** [comment]
I'm sorry, I don't understand your respond. Indeed the type of fields is a tuple of fields. My issue is that the type of `field.type` does not match the typeshed declaration. I suppose that now that `Any` is in the type, there's no point in having `type[T]` in the union (nor `str` for that matter), but it's still jarring that pyright's type does not match the typeshed declaration.

**Comment by erictraut** [comment]
Ah, I see what you mean now. You expected the resulting type to be `type[Any] | str | Any`. 

Here's a code sample that demonstrates this more clearly.

```python
from dataclasses import Field
from typing import Any

def func(x: Field[Any]):
    reveal_type(x.type) # should be type[Any] | str | Any
```

I agree this is a bug, and it will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.372.
